v0/customerio/transform.js

```diff
@@ -0,0 +1,222 @@
+      eventProps.forEach(prop => {
+        let val = get(message, "properties." + prop);
+        if (isPrimitive(val)) {
+          set(rawPayload.data, prop, val);
+        }
+      }); */
+
+      if (deviceDeleteRelatedEventName == evName) {
+        const token = get(message, "context.device.token");
+        if (userId && token) {
+          endpoint = DEVICE_DELETE_ENDPOINT.replace(":id", userId).replace(
+            ":device_id",
+            token
+          );
+
+          response.endpoint = endpoint;
+          response.method = "DELETE";
+          response.headers = {
+            Authorization:
+              "Basic " +
+              btoa(destination.Config.siteID + ":" + destination.Config.apiKey)
+          };
+
+          return response;
+        } else {
+          return {
+            statusCode: 400,
+            error: "userId or device_token not present"
+          };
+        }
+      }
+
+      if (deviceRelatedEventNames.includes(evName)) {
+        let devProps = message.properties;
+        set(devProps, "id", get(message, "context.device.token"));
+        set(devProps, "platform", get(message, "context.device.type"));
+        set(
+          devProps,
+          "last_used",
+          Math.floor(new Date(message.originalTimestamp).getTime() / 1000)
+        );
+        set(rawPayload, "device", devProps);
+        requestConfig = defaultPutRequestConfig;
+      } else {
+        rawPayload.data = {};
+        set(rawPayload, "data", message.properties);
+      }
+    }
+
+    if (!deviceRelatedEventNames.includes(evName)) {
+      set(rawPayload, "name", evName);
+      set(rawPayload, "type", evType);
+    }
+
+    if (userId) {
+      if (deviceRelatedEventNames.includes(evName)) {
+        endpoint = DEVICE_REGISTER_ENDPOINT.replace(":id", userId);
+      } else {
+        endpoint = USER_EVENT_ENDPOINT.replace(":id", userId);
+      }
+    } else {
+      endpoint = ANON_EVENT_ENDPOINT;
+    }
+  }
+  const payload = removeUndefinedValues(rawPayload);
+  //console.log(payload);
+  response.endpoint = endpoint;
+  response.method = requestConfig.requestMethod;
+  response.headers = {
+    "Content-Type": "application/json",
+    Authorization:
+      "Basic " +
+      btoa(destination.Config.siteID + ":" + destination.Config.apiKey)
+  };
+  response.body.JSON = payload;
+
+  //console.log(response);
+  return response;
+}
+
+function processSingleMessage(message, destination) {
+  let messageType = message.type.toLowerCase();
+  let evType;
+  let evName;
+  //console.log("messageType", messageType);
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      evType = "identify";
+      break;
+    case EventType.PAGE:
+      evType = "page"; // customerio mandates sending 'page' for pageview events
+      evName = message.name ? message.name : message.properties.url;
+      break;
+    case EventType.SCREEN:
+      evType = "event";
+      evName =
+        "Viewed " + message.name
```

Comment:
Can we put the part after Viewed in ().
Also since JS has no type, we could merge some ternary operator to || statements.

Reply:
Updated